tests/test_picking_batch.py

```diff
@@ -305,3 +307,215 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
+            self.assertEqual(err.exception.name,
+                             'Cannot mark a move line as unpickable '
+                             'when it is part of a completed Picking')
+
+    def test20_unpickable_item_invalid_state_done(self):
+        """
+        Tests that a ValidationError is raised if the picking is on a state of
+        done
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.update_picking(force_validate=True,
+                               location_dest_id=self.test_output_location_01.id)
+
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=move_line_id,
+                                  reason=reason,
+                                  picking_type_id=None)
+            self.assertEqual(err.exception.name,
+                             'Cannot mark a move line as unpickable '
+                             'when it is part of a completed Picking')
+
+    def test18_unpickable_item_no_package_vaidation_error(self):
+        """
+        Tests that ValidationError is raised if the move_line does not have
+        a package.  This functionality is not yet handled by the system.
+        :return:
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        reason = 'missing item'
+        move_line = picking.move_line_ids[0]
+        # Make the move_line not have a package.
+        move_line.package_id = None
+        with self.assertRaises(ValidationError) as err:
+            batch.unpickable_item(move_line_id=move_line.id,
+                                  reason=reason,
+                                  picking_type_id=None)
+
+            self.assertEqual(err.exception.name,
+                             'Not Implemented')
+
+    def test19_unpickable_item_multiple_move_lines_different_packages(self):
+        """
+        Tests that a backorder is created and cancelled if there are multiple
+        move lines on the picking.  The original picking should continue to
+        have the still pickable product on it.
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_one.id)
+        self.create_quant(self.banana.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        products_info = [{'product': self.apple,
+                          'qty': 4},
+                         {'product': self.banana,
+                          'qty': 4}]
+        picking = self.create_picking(self.picking_type_pick,
+                                      products_info=products_info,
+                                      confirm=True,
+                                      assign=True)
+        Batch = self.env['stock.picking.batch']
```

Comment:
could you move this definition to the top of the method, just for style consistency?